portable/ThirdParty/GCC/Posix/port.c

```diff
@@ -281,10 +282,17 @@ BaseType_t xPortStartScheduler( void )
      */
     xSchedulerEnd = pdFALSE;
 
-    /* Reset the pthread_once_t structure. This is required if the port
-     * starts the scheduler again. */
-    hSigSetupThread = PTHREAD_ONCE_INIT;
-
+    /* Reset pthread_once_t, needed to restart the scheduler again.
+     * memset the internal struct members for MacOS/Linux Compatability */
+    #if __APPLE__
+        /* Restarting the scheduler is currently unsupported on MacOS. */
+        configASSERT( pdFALSE );
+        hSigSetupThread.__sig = _PTHREAD_ONCE_SIG_init;
+        memset( ( void * ) &hSigSetupThread.__opaque, 0, sizeof(hSigSetupThread.__opaque));
```

Comment:
A posix port running on mac-OS can still call `vTaskEndScheduler()` to stop the scheduler. Suggest we just keep the comment and remove these lines since restarting scheduler is not supported in mac-OS. We can consider to support the feature in mac-OS with future commit.

Reply:
Calling `vTaskEndScheduler()/vTaskStartScheduler()`in the [Posix_GCC Demo](https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Demo/Posix_GCC) was leading to a segfault when I called it on Linux. Where when calling it in MacOS I was seeing the scheduler thread immediately dying after the call to `( void ) pthread_sigmask( SIG_SETMASK, &xSchedulerOriginalSignalMask, NULL );` on the following line.

Since I didn't have a working example to compare to I added the `configASSERT()`. I believe this would be a more "user friendly" way of learning that this feature is not supported on MacOS compare to having a seemingly paused system due to the scheduler thread being killed.

Reply:
Restarting FreeRTOS kernel by calling `vTaskStartScheduler()` again after `vTaskEndScheduler()` is a new feature. Resetting kernel state is required before next time calling `vTaskStartScheduler()` again to restart scheduler. A PR https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/944 to add reset state function is created for discussion. The problem to restart scheduler in Linux or MacOS may be related to kernel state is not reset to their default value.

Posix port should not have problem to stop scheduler with `vTaskEndScheduler()` for the first time. User may call `vTaskEndScheduler()` for their testing purpose. For example, in this PR https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/728 user is fixing MSVC-port end scheduler function to run this port with GoogleTest. The `configASSERT` added may cause problem to users who want to stop the scheduler.

Reply:
I have removed the lines
I was running the [examples/tests](https://github.com/cmorganBE/freertos_posix) linked in #914 and was seeing segfaults on Linux. This lead me to believe this feature was unsupported, and why I added a configASSERT() instead of segfaulting.

I have run the [POSIX_GCC Blinky Demo](https://github.com/FreeRTOS/FreeRTOS/blob/d603c9b1ba39a5f1c1ce8ea42397c72cc5bfcaa3/FreeRTOS/Demo/Posix_GCC/main_blinky.c#L1) and called `vTaskEndScheduler()` on MacOS to validate these changes.